invenio_app_ils/records/jsonresolver/document.py

```diff
@@ -27,36 +25,20 @@ def jsonresolver_loader(url_map):
     def circulation_resolver(document_pid):
         """Return circulation info for the given Document."""
         circulation = {}
-        active_loans_count = 0
-        loanable_items_count = 0
-        items_count = 0
-        item_pids = circulation_items_retriever(document_pid)
 
-        for pid in item_pids:
-            items_count += 1
-            if not is_item_available(pid):
-                active_loans_count += 1
+        items = list(ItemSearch().search_by_document_pid(document_pid).scan())
+        loan_search = IlsLoansSearch()
 
-            if circulation_is_item_available(pid):
-                loanable_items_count += 1
+        past_loans_count = loan_search.get_past_loans_by_doc_pid(document_pid).execute().hits.total
+        active_loans_count = loan_search.get_active_loans_by_doc_pid(document_pid).execute().hits.total
+        pending_loans_count = loan_search.get_pending_loans_by_doc_pid(document_pid).execute().hits.total
 
-        search = search_by_pid(
-            document_pid=document_pid,
-            filter_states=current_app.config.get(
-                "CIRCULATION_STATES_LOAN_COMPLETED", []
-            ),
-        )
-
-        pending_loans = search_by_pid(document_pid=document_pid,
-                                      filter_states=['PENDING']
-                                      ).execute().hits.total
-
-        circulation["number_of_past_loans"] = search.execute().hits.total
-        circulation["number_of_items"] = items_count
+        circulation["number_of_past_loans"] = past_loans_count
+        circulation["number_of_items"] = len(items)
         circulation["active_loans"] = active_loans_count
-        circulation["loanable_items"] = loanable_items_count
-        circulation["pending_loans"] = pending_loans
-        circulation["overbooked"] = pending_loans > loanable_items_count
+        circulation["loanable_items"] = sum(item["status"] == "LOANABLE" for item in items)
```

Comment:
the logic of this statistic is a bit wrong, because we want to count the items which are not on loan currently also, not only `LOANABLE`, that's why there was `is_item_available` function